### DISCUSSION TITLE
How record a input stream only when is live?

### DISCUSSION THREAD

**Discussion by juancrobles** [savonet/liquidsoap#2567] [Q&A]
I have a script that switches between a live stream for a couple of hours and a sequential playlist, I like to dump to file, the input live for the time that is live instead of dump the output stream all the time

**Comment by vitoyucepi** [comment]
Hi, @juancrobles.
Let's say you have
```ruby
live = input.harbor(
  port = 8000,
  user = 'source',
  password = 'hackme',
  '/live'
)
```
Then you can
```ruby
output.file(%vorbis, '/path/to/output.ogg', live, fallible = true)
```

Check this sample https://www.liquidsoap.info/doc-2.1.0/radiopi.html.
Also you could check page 128 of [the book](https://www.liquidsoap.info/book/book.pdf).

**Reply by juancrobles** [reply]
ok, that helps, minor issue how can "trim" the live stream before stored it to the file?, my situation is that the stream start at certain time but it does not go to the output stream until ~20 mins later

**Reply by vitoyucepi** [reply]
It depends on how stream behaves during initial phase.
Maybe https://www.liquidsoap.info/doc-2.1.0/reference.html#blank.strip operator will help you.

**Reply by juancrobles** [reply]
it can have, let say 5 mins of talk and then 3 of silence and talk again before the proper show... that operator I'm using it to detect the end of the live stream to switch to the playlist

**Reply by vitoyucepi** [reply]
It's hard for me to say how to do it properly.
Could you describe your pipeline in details?

**Reply by juancrobles** [reply]
my objective is, build an automated stream which is composed of a live feed about 3 hours and a playlist of past shows. Of the live feed one of the three hours is pre-show chat between the participants the other two are the show.
currently the pipeline is through audio hijack directly streaming to icecast

**Reply by vitoyucepi** [reply]
@juancrobles Not so much details.

Well, could you draw a diagrams using https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams.
For example you could draw a [gantt](https://mermaid-js.github.io/mermaid/#/gantt) diagram of your live events.

Would you mind posting some of your liquidsoap configs, because at the moment I have no clue how do you pipeline the stream?
